version: "3"

services:
    proxy:
        image: nginx:1.19.7-alpine
        container_name: back-challenge-proxy
        depends_on:
            - api
        ports:
            - "80:80"
        networks:
            - backchallengenet
        volumes:
            - ./nginx/templates:/etc/nginx/templates

    api:
        image: backchallenge/api
        container_name: back-challenge-api
        build:
            context: ..
            dockerfile: ./infrastructure/django/Dockerfile
        volumes:
            - ../:/opt/
        env_file:
            - ../.env
        networks:
            - backchallengenet
        expose:
            - 5000
        depends_on:
            - "db"

    db:
        image: mysql:5.7.22
        container_name: back-challenge-db
        restart: unless-stopped
        tty: true
        ports:
            - "3306:3306"
        environment:
            MYSQL_DATABASE: BACK_CHALLENGE
            MYSQL_ROOT_PASSWORD: 12345678
        volumes:
            - back-challengedbdata:/var/lib/mysql
            - ./mysql/my.cnf:/etc/mysql/my.cnf
            - ./mysql/back_challenge.sql:/tmp/back_challenge.sql
        networks:
            - backchallengenet

networks:
    backchallengenet:

volumes:
    back-challengedbdata:
        driver: local
